packages/@aws-cdk/cdk/test/core/test.tag-manager.ts

```diff
@@ -0,0 +1,163 @@
+import { Test } from 'nodeunit';
+import { Construct, Root } from '../../lib/core/construct';
+import { ITaggable, ITags, TagManager } from '../../lib/core/tag-manager';
+
+class ChildTagger extends Construct implements ITaggable {
```

Comment:
Add a use case where only a grand-grand(-grand?)-parent is taggable. I still expect the grand-grand-children to get propagated tags.